### PR TITLE
DnD fixes

### PR DESCRIPTION
**Pull request by mkoty**

### CODE REVIEW

File: src/common/dnd/drag.directive.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ export class DragDirective implements OnInit, OnDestroy {
 
   start(e) {
     const transfer = e.dataTransfer;
-    if (this.canDrag(this.event()) === false) {
+    if (this.canDrag === false) {
```
Comment (klumba12) [inline src/common/dnd/drag.directive.ts]: Shouldn't be it a function?

Comment (klumba12) [inline src/common/dnd/drag.directive.ts]: is'it bindined automatically?

File: src/main/core/head/head-core.component.html [hunk 2]
```diff
@@ -6,9 +6,9 @@
       [attr.colspan]="$column.colspan"
       [q-grid-drag]="$view.head.transfer($column.model)"
       [q-grid-drop]="$view.head.transfer($column.model)"
-      q-grid-on-drop="$view.head.drop.execute($event)"
-      q-grid-can-drop="$view.head.drop.canExecute($event)"
-      q-grid-can-drag="$view.head.drag.canExecute($event)"
+      (q-grid-on-drop)="$view.head.drop.execute($event)"
+      [q-grid-can-drop]="$view.head.drop.canExecute($column.model)"
```
Comment (klumba12) [inline src/main/core/head/head-core.component.html]: I thinks this not very good, cause it'll trigger every time on change tracker we need to pass it as a function

File: core/head/head.view.js [hunk 3]
```diff
@@ -12,13 +12,9 @@ export class HeadView extends View {
 
 		this.drop = new Command({
 			canExecute: e => {
-				if (e.source && e.source.key === tagName) {
```
Comment (klumba12) [inline core/head/head.view.js]: Don't we need to check the tag name? It's kinda of convension to check if source of drag/drop is valid

File: src/common/dnd/drag.directive.ts (hunk 1 above)
Comment (mkoty) [inline src/common/dnd/drag.directive.ts]: from head.view.js:
canDrag = `map.hasOwnProperty(e.key)` && map[e.key].canMove !== false`
It's just boolean value

File: core/head/head.view.js (hunk 3 above)
Comment (mkoty) [inline core/head/head.view.js]: In that moment: `e.source:"user"`

File: src/common/dnd/drop.directive.ts [hunk 4]
```diff
@@ -27,6 +29,7 @@ export class DropDirective implements OnInit, OnDestroy {
     this.listener.on('dragover', this.over);
     this.listener.on('dragleave', this.leave);
     this.listener.on('drop', this.drop);
+    this.canDrop = this.canDropFunction(this.model);
```
Comment (klumba12) [inline src/common/dnd/drop.directive.ts]: i think this should be dynamic, depending on dragging stuff so it should be a function

File: src/common/dnd/drag.directive.ts (hunk 1 above)
Comment (klumba12) [inline src/common/dnd/drag.directive.ts]: It can be dynamic depending on grid state so it should be a function

**Pull request merged**